stacker/plan.py

```diff
@@ -150,8 +152,16 @@ def add(self, stack, run_func, requires=None):
             stack=stack,
             run_func=run_func,
             requires=requires,
+            tail=self.tail
         )
 
+    def poll(self):
+        stack_dict = self._poll_func(self.tail)
+        for step_name, step in self.list_pending():
+            if step_name in stack_dict:
```

Comment:
I'd replace this with:

```
try:
    status = stack_dict[step_name]
except KeyError:
    continue
step.set_status(status)
```

Usually it's better to just try and use something that should be there, then fail gracefully if it doesn't exist, rather than testing to see if it's there and then doing the work.